### DISCUSSION TITLE
Real-time trajectory tracking

### DISCUSSION THREAD

**Discussion by sadlim** [USC-ACTLab/crazyswarm#689] [Q&A] [answered]
Hi,
I want to implement real-time trajectory tracking, i.e. calculate the next point by MatLab, use the `goTo` command to make crazyflie fly to that point, then Matlab calculates the next point according to crazyflie's current position, then send it to crazyflie, and so on. 

But this will cause the plane to fly to the desired point and then the **velocity becomes zero** and the trajectory is not smooth.

Is there any way to solve this problem? Or is there any more suitable way to reach my initial purpose, like using `cmdFullState()`? 

Looking forward to and appreciate your answers.

**Comment by jpreiss** [comment, marked as answer]
The idea of `goTo` with nonzero final velocity raises some questions about how the commander should behave after the trajectory has ended.

Should the commander continue to maintain the same velocity forever? Or should there be some kind of timeout period?

When maintaining velocity after a `goTo` trajectory has ended, should the controller still attempt to track the position `final_velocity * time_since_trajectory_ended`? Or should we switch to a velocity-only controller?

So far, nobody has needed this feature enough to try to resolve these design decisions and do the implementation work.

You can certainly achieve your goal with low-level streaming modes like `cmdFullState`, `cmdVelocityWorld`, and `cmdPosition`, but you will need to send setpoints quickly (around 10Hz for position-only, faster if you set nonzero angular velocity in the `cmdFullState`).

BTW, this question is not Crazyswarm-specific. You may get more answers on the Bitcraze discussions.

**Reply by sadlim** [reply]
Thank u, I'll think about it~
